### PR TITLE
Add price to assets home private asset packs thumbnails and display them in their dialog

### PR DESCRIPTION
**Pull request by AlexandreSi**
Create a new component to display price => [story](https://gdevelop-storybook.s3.amazonaws.com/add-private-asset-packs-prices/latest/index.html?path=/story/ui-building-blocks-pricetag--default)

image

image

### CODE REVIEW

File: newIDE/app/src/UI/PriceTag.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import Text from './Text';
+import { makeStyles } from '@material-ui/core';
+
+type Props = {|
+  value: number,
+  /**
+   * To be used when the component is over an element for which
+   * we don't control the background (e.g. an image).
+   * This component should logically be invariant to selected theme
+   * when this option is activated
```
Comment (4ian) [inline newIDE/app/src/UI/PriceTag.js:15]: Not sure I totally understood the comment

File: newIDE/app/src/UI/BackgroundText.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ const BackgroundText = (props: Props) => (
       <Typography
         variant="body2"
         align="center"
+        component="div"
```
Comment (4ian) [inline newIDE/app/src/UI/BackgroundText.js:19]: Nice!

**Comment by D8H** [comment]
How does the description looks on a small screen?

File: newIDE/app/src/UI/PriceTag.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      {({ i18n }) => (
+        <div className={classes.container}>
+          <Text noMargin size="sub-title" color="inherit">
+            {i18n
+              .number(value / 100, {
+                minimumFractionDigits: 2,
+                maximumFractionDigits: 2,
+              })
+              .replace(/\D00$/, '')}
+            EUR
```
Comment (4ian) [inline newIDE/app/src/UI/PriceTag.js:56]: I think the order is usually EUR 100. We could also use the symbol € => €10?

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/UI/PriceTag.js (hunk 3 above)
Comment (AlexandreSi) [reply newIDE/app/src/UI/PriceTag.js:56]: This is for @LuniMoon ;)

File: newIDE/app/src/UI/PriceTag.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/UI/PriceTag.js:15]: I'll clarify

**Comment by AlexandreSi** [comment]
If you want to make your own tests => https://gdevelop-storybook.s3.amazonaws.com/add-private-asset-packs-prices/latest/index.html?path=/story/assetstore-assetstore-privateassetpackdialog--default

Otherwise, here it is:
image

File: newIDE/app/src/AssetStore/index.js [hunk 4]
```diff
@@ -300,6 +302,16 @@ export const AssetStore = ({ project }: Props) => {
                     }
                   />
                 )}
+                {isOnHomePage && error && (
+                  <PlaceholderError>
```
Comment (ClementPasteau) [inline newIDE/app/src/AssetStore/index.js:306]: should we pass an `onRetry` to the component with the function `fetchAssetsAndFilters`

File: newIDE/app/src/UI/PriceTag.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      borderRadius: 4,
+      padding: '2px 4px',
+      backdropFilter: props =>
+        props.withOverlay && theme.palette.type === 'light'
+          ? 'brightness(40%)'
+          : undefined,
+      backgroundColor: props =>
+        props.withOverlay && theme.palette.type === 'light'
+          ? undefined
+          : theme.palette.background.alternate.startsWith('#')
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/PriceTag.js:37]: hmmm, that feels like a hack 😅 
Should we try to have all colors in the same format in the themes (start with '#')?

File: newIDE/app/src/UI/PriceTag.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          ? theme.palette.background.alternate + 'BB' // manually add opacity to the background hex color
+          : theme.palette.background.alternate,
+      color: props =>
+        props.withOverlay && theme.palette.type === 'light'
+          ? '#FAFAFA'
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/PriceTag.js:42]: I don't mind having a color here, but wondering if we should avoid that and put everything in the theme.json files?

**Review by ClementPasteau — APPROVED** [review]
few comments/questions, but lgtm!

File: newIDE/app/src/UI/PriceTag.js (hunk 3 above)
Comment (ClementPasteau) [reply newIDE/app/src/UI/PriceTag.js:61]: Agree that the opposite order is what I've been used to, but to check with Luni

File: newIDE/app/src/AssetStore/index.js (hunk 4 above)
Comment (AlexandreSi) [reply newIDE/app/src/AssetStore/index.js:306]: Good idea! I'll add it

File: newIDE/app/src/UI/PriceTag.js (hunk 3 above)
Comment (LuniMoon) [reply newIDE/app/src/UI/PriceTag.js:61]: Sorry! that price got lost in translation hehe.

I've just bench a little here and there.

**Amazon**:
Screenshot 2022-10-04 at 11 00 37

**Google Search** results:
Screenshot 2022-10-04 at 11 03 31

**United Nude** (shoes):
Screenshot 2022-10-04 at 11 02 40

**Quicksilver** (clothes):
Screenshot 2022-10-04 at 11 03 11

**Vans** (clothes):
Screenshot 2022-10-04 at 11 04 13

**KLM** (flights)
Screenshot 2022-10-04 at 11 05 01

**I love NY** official store:
Screenshot 2022-10-04 at 11 10 49


So, let's go with the majority, and **keep the "€" at the beginning of the price**.
As for the decimals, we'll use 2 (if needed), and I personally have a preference for a dot separating the cents from the units (I'm from the american school), but I dunno if that's good practice in Europe or the rest of the world (?).
I do see sites using commas and dots interchangeably...

Comment (LuniMoon) [reply newIDE/app/src/UI/PriceTag.js:61]: Note on the side: In my context, **commas are used for big numbers like "2,599" and dots for cents "2.59"**.

But I do know that depending on the language, numbers are written differently (each language has certains rules for using commas and dots, and sometimes they vary from each other... I do have some books as reference if you wanna read about it lol).

Comment (AlexandreSi) [reply newIDE/app/src/UI/PriceTag.js:61]: > I personally have a preference for a dot separating the cents from the units

This should be handled by the native internationalization library.
In French, this is really weird for non-english speaking people to see a price like `1,255.45`. I guess other countries have the same kind of issues.

**Pull request merged**